test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -524,7 +524,7 @@ private static Settings getRandomNodeSettings(long seed) {
 
         builder.put(
             EsExecutors.NODE_PROCESSORS_SETTING.getKey(),
-            1 + random.nextInt(Math.min(4, Runtime.getRuntime().availableProcessors()))
+            RandomNumbers.randomIntBetween(random, 1, Runtime.getRuntime().availableProcessors())
```

Comment:
Ok, so looks like we were _already_ doing this sort of thing for internal cluster tests.